docs/designers-developers/developers/tutorials/javascript/readme.md

```diff
@@ -0,0 +1,108 @@
+```
+
+Create a file called `myguten.js` and add:
+
+```js
+console.log( "I'm loaded!" );
+```
+
+Now go to any page within `/wp-admin/` and check your browser Developer Tools, and you should see the message in your console log. See [Mozilla's What are browser developer tools?](https://developer.mozilla.org/en-US/docs/Learn/Common_questions/What_are_browser_developer_tools) if you need more information, the area to become most familiar with is [The JavaScript console](https://developer.mozilla.org/en-US/docs/Learn/Common_questions/What_are_browser_developer_tools#The_JavaScript_console).
+
+The script enqueuing used the `admin_enqueue_scripts` hook which only loads the JavaScript within the wp-admin section. So if you navigate to any front-end page or article on your site, you will not see the message.
+
+**Note:** The last argument in the `wp_enqueue_script()` function is an array of dependencies, all of the Gutenberg packages are registered and can be loaded by specifying them in the array, blocks and elements are shown as two common examples.
+
+Recap at this point, we have a plugin which loads JavaScript, we're off to a good start.
+
+
+## JavaScript versions and building
+
+The Gutenberg Handbook shows examples in two syntaxes ES5 and ESNext, note ESNext is sometimes referred to as ES6. The ES5 code is compatible with almost all current browsers.
+
+The ESNext syntax is compatible with most modern browsers, but unfortunately not IE11. Additionally, the ESNext code examples include JSX syntax, which requires a build step using webpack to transform into code that can be run in your browser.
+
+For this tutorial, all examples will be in straight ES5 code that can be run straight in your browser and does not require an additional build step.
+
+
+## Extending the Block Editor
+
+This puts all the initial pieces in place for you to start extending the Block Editor.
+
+Let's look at using the [Block Style Variation example](../../../../../docs/designers-developers/developers/filters/block-filters/#block-style-variations).
+
+Update the file `myguten.js` with:
+
+```js
+wp.blocks.registerBlockStyle( 'core/quote', {
+    name: 'fancy-quote',
+    label: 'Fancy Quote'
+} );
+```
+
+You can remove the previous `console.log` if you like.
+
+After you add the code, save the file, and start a new post in the Block Editor.
+
+Add a quote block, and in the right sidebar under Styles, you should see your new Fancy Quote style listed. You can go back to the JavaScript and change the label to "Fancy Pants" and reload, and you should see the new label.
+
+Previewing or Publishing the post, you will not see a visible change. However, if you look at the source, you should see `is-style-fancy-quote` class name attached to your quote.
+
+You could create a `style.css` file with:
+
+```css
+.is-style-fancy-quote {
+	font-size: 64px;
+}
+
+```
+
+and enqueue the CSS by adding the following to your `myguten-plugin.php`:
+
+```php
+function myguten_stylesheet() {
```

Comment:
Can this function have PHPDoc too, explaining what it's doing?

Reply:
I've broken up the code and changed things around with more explanations above the code. My concern is if it were as a comment in code people will miss it. I've also included links to all referred to functions for any additional details are available.